extensions/guacamole-vault/modules/guacamole-vault-hv/src/main/java/org/apache/guacamole/vault/hv/secret/HvSecretService.java

```diff
@@ -0,0 +1,335 @@
+            ConnectionGroup group = connectionGroupDirectory.get(parentIdentifier);
+            if (group == null)
+                break;
+
+            // If the current connection group has the HV configuration attribute
+            // set to a non-empty value, return immediately
+            String hvConfig = group.getAttributes().get(HvAttributeService.HV_CONFIGURATION_ATTRIBUTE);
+            if (hvConfig != null && !hvConfig.trim().isEmpty())
+                return hvConfig;
+
+            // Otherwise, keep searching up the tree until an appropriate configuration is found
+            parentIdentifier = group.getParentIdentifier();
+
+            // If the parent is a group that's already been seen, this is a cycle, so there's no
+            // need to search any further
+            if (!observedIdentifiers.add(parentIdentifier))
+                break;
+        }
+
+
+        // If no HV configuration was ever found, use the default value
+        return confService.getHvConfig();
+    }
+
+    /**
+     * Returns true if user-level HV configuration is enabled for the given
+     * Connectable, false otherwise.
+     *
+     * @param connectable
+     *     The connectable to check for whether user-level HV configs are
+     *     enabled.
+     *
+     * @return
+     *     True if user-level HV configuration is enabled for the given
+     *     Connectable, false otherwise.
+     */
+    private boolean isHvUserConfigEnabled(Connectable connectable) {
+
+        // User-level config is enabled IFF the appropriate attribute is set to true
+        if (connectable instanceof Attributes)
+            return HvAttributeService.TRUTH_VALUE.equals(((Attributes) connectable).getAttributes().get(
+                HvAttributeService.HV_USER_CONFIG_ENABLED_ATTRIBUTE));
+
+        // If there's no attributes to check, the user config cannot be enabled
+        return false;
+
+    }
+
+    /**
+     * Return the HV config blob for the current user IFF user HV configs
+     * are enabled globally, and are enabled for the given connectable. If no
+     * HV config exists for the given user or HV configs are not enabled,
+     * null will be returned.
+     *
+     * @param userContext
+     *    The user context from which the current user should be fetched.
+     *
+     * @param connectable
+     *    The connectable to which the connection is being established. This
+     *    is the connection which will be checked to see if user HV configs
+     *    are enabled.
+     *
+     * @return
+     *    The base64 encoded HV config blob for the current user if one
+     *    exists, and if user HV configs are enabled globally and for the
+     *    provided connectable.
+     *
+     * @throws GuacamoleException
+     *     If an error occurs while attempting to fetch the HV config.
+     */
+    private String getUserHVConfig(UserContext userContext, Connectable connectable) throws GuacamoleException {
+        return null;
+    }
```

Comment:
First, the `throws...` part of this should probably be on a separate line. But, also, this doesn't seem to do any of the things documented in the function? Not sure if you just didn't get around to implementing it, or if this is just a stub that might be overridden elsewhere? Either the `return null;` line should document that, or the function should be updated or removed?